src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2192,7 +2192,15 @@ static std::optional<CoordsXY> ride_get_place_position_from_screen_position(Scre
     else
     {
         auto mapZ = _trackPlaceCtrlZ;
-        mapCoords = screen_get_map_xy_with_z(screenCoords, mapZ);
+        auto mapXYCoords = screen_get_map_xy_with_z(screenCoords, mapZ);
+        if (mapXYCoords)
+        {
+            mapCoords = *mapXYCoords;
+        }
+        else
+        {
+            mapCoords.x = LOCATION_NULL;
```

Comment:
return a nullopt since thats all that is going to happen after this section of code.